contracts/src/messaging/messenger/CrossChainMessenger.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./ICrossChainMessenger.sol";
+
+contract CrossChainMessenger is ICrossChainMessenger {
```

Comment:
this needs a significant comment.
Explaining what is its role, what is the intended setup. 
who will own it.